### DISCUSSION TITLE
Timestamp issues with using --language English to force transcription of non-English audio.

### DISCUSSION THREAD

**Discussion by despairTK** [Purfview/whisper-standalone-win#129] [Q&A] [answered]
This problem is mainly for transcribing non-English audio. I have tested Japanese audio, Spanish audio, and Korean audio. If the language specified when generating subtitles is the same as the language of the transcribed audio (such as Japanese audio, set --language Japanese), the subtitle results will be There will be some selection errors for polysyllabic words. However, if you force the generated subtitles to be in English (such as Japanese audio, set --language English), you will get more accurate subtitle results than the original language of the audio. And the results obtained by using --language English to force the transcription of non-English audio are very different from --task translate, and are more accurate than --task translate. However, the timestamp may have incorrect positions or be of too short duration, and there may also be repeated hallucinations.

It would be even better if the timestamps could be normal when using --language English to force transcribe non-English audio. Otherwise, you can only manually correct the timestamp of each line of subtitles.

Hallucinations and timestamp issues that may occur:
![QQ截图20231202100137](https://github.com/Purfview/whisper-standalone-win/assets/111871110/3128f507-c29a-4ee7-b584-75e5388ad6bc)

**Comment by Purfview** [comment, marked as answer]
Not so accurate timestamps on the translations is known thing. It think I even added a warning about it.

I've no idea about your observations  with `--language English` vs proper language and `--task=translate`.
Better ask there -> https://github.com/openai/whisper/discussions

**Reply by despairTK** [reply]
OK, thank you very much for your answer.

Overall, my translation abilities are: Japanese>English>Korean>Other languages. It’s not that I don’t understand English, it’s just that I can’t better express some professional words in the professional field when I feedback some questions to you. After all, I rarely come into contact with such professional words in daily translation. I can still understand ordinary, common sentences and words very well.
